Tools/ssl/multissltests.py

```diff
@@ -160,30 +127,55 @@
 )
 
 
-class AbstractBuilder(object):
-    library = None
-    url_templates = None
-    src_template = None
-    build_template = None
+class AbstractBuilder(object, metaclass=ABCMeta):
     depend_target = None
     install_target = 'install'
     if hasattr(os, 'process_cpu_count'):
         jobs = os.process_cpu_count()
     else:
         jobs = os.cpu_count()
-
     module_files = (
         os.path.join(PYTHONROOT, "Modules/_ssl.c"),
         os.path.join(PYTHONROOT, "Modules/_hashopenssl.c"),
     )
     module_libs = ("_ssl", "_hashlib")
 
+    @property
+    @abstractmethod
+    def library(self=None):
+        pass
```

Comment:
Please revert this. As we can't have abstract class methods, I prefer having real class variables where we would raise if they are not set.